src/v1.rs

```diff
@@ -226,21 +226,36 @@ impl Party {
         signers: &[u32],
         nonces: &[PublicNonce],
         aggregate_nonce: &Point,
-        tweak: &Scalar,
+        tweak: Option<Scalar>,
     ) -> SignatureShare {
         let mut r = &self.nonce.d + &self.nonce.e * compute::binding(&self.id(), nonces, msg);
-        if tweak != &Scalar::zero() && !aggregate_nonce.has_even_y() {
+        if tweak.is_some() && !aggregate_nonce.has_even_y() {
```

Comment:
I think we should add more comments in this function, and/or link to some artifact that specifies what logic this function is supposed to be following.

Same thing for `Aggregator::sign_with_tweak`, `Aggregator::sign_schnorr`, and `Aggregator::sign_taproot`. Not sure if `Aggregator::check_signature_shares` should be put in that camp as well. And we probably want to add this for the v2 versions as well.

Reply:
Oh some of the functions I mentioned above are trait methods, so maybe specify that with the trait.

Reply:
Done